docs/DeveloperGuide.md

```diff
@@ -1,34 +1,513 @@
+
+<small><i>Figure 4</i></small>
+
+1. `Logic` stores a current `Template` object in [`Model`](#model-component) that represents the current circuit configuration.
+1. `Logic` uses the `Parser` class to parse the user command.
+1. This results in a `Command` object which is executed in `Duke`.
+1. The command execution can affect the `Model` (e.g. setting a value).
+1. In addition, the `Ui` may also perform certain actions, such as displaying help to the user.
+
+### Model component
+
+![ModelDiagram](diagrams/ModelClassDiagram.png)
+
+<small><i>Figure 5</i></small>
+
+The `Model`,
+* includes `CircuitTemplate` and `BooleanTemplate` that can represent the current `template` in [`Logic`](#logic-component).
+* has `Component` and `Gate` within the templates.
+* does not depend on any of the other three components.
+
+--------------------------------------------------------------------------------------------------------------------
+
+## Implementation of Circuit Commands (Dash) <a name="circ-comd"></a>
+
+This section provides details on the implementation of the various electronic circuit commands.
+
+There are 4 different types of components of electronic circuits that can be instantiated in the program:
+* `Resistor` - A _resistor_ component.
+* `Capacitor` - A _capacitor_ component.
+* `Inductor` - An _inductor_ component.
+* `VoltageSource` - An _Alternating Current_ Voltage Source.
+
+Each component is used within a circuit template, based on the kind of circuit instantiated.
+
+<div markdown="span" class="alert alert-primary">
+
+:information_source: **Note:** The `VoltageSource` is instantiated in all Circuit Templates.
+
+</div>
+
+There are four different circuit templates that can be instantiated in the program:
+
+* `LcTemplate` - Inductor-Capacitor Circuit Template
+* `RTemplate` - Resistor Circuit Template
+* `RcTemplate` - Resistor-Capacitor Template (extends `RTemplate`)
+* `LrTemplate` - Inductor-Resistor Template (extends `RTemplate`)
+
+![CircuitCommandClass](diagrams/CircuitCommandClassDiagram.png)
+
+<small><i>Figure 6</i></small>
+
+The diagram above demonstrates the relationship between the various `CircuitCommand` objects. The various commands to be parsed are as explained in this section. While the **User Guide** explains the commands used on the CLI, this section goes into detail the classes used to execute the commands. The command classes also make use of the classes in the [**`Model`**](#model-component) shown in the diagram below when executing the commands.
+
+![CircuitModelClass](diagrams/CircuitModelClassDiagram.png)
+
+<small><i>Figure 7</i></small>
+
+### `TemplateCircuitCommand`  <a name="temp-circ"></a>
+Given below is the sequence diagram for interactions within the `logic` and `template` components for the 
+`parse(template rl)` API call that implements the `template` command to create templates.
+
+![TemplateSequence](diagrams/TemplateSequenceDiagram.png)
+
+<small><i>Figure 8</i></small>
```

Comment:
Maybe extend the the activation bar of the parser object such that command return is extended from the bar's edge instead of from the lifeline itself.

Reply:
Fixed in #119.